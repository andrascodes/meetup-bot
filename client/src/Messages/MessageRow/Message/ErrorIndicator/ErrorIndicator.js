import React from 'react'

import './ErrorIndicator.css'

export const ErrorIndicator = props => {
  if(!props.value) {
    return (<div></div>)
  }

  return (
    <div className="ErrorIndicator">
      {`Error: `}
      <span className="ErrorStyle">{`${props.value} #${props.response.error.code}`}</span>
      <i className="fa fa-times ErrorStyle Icon"></i>
    </div>
  )
}