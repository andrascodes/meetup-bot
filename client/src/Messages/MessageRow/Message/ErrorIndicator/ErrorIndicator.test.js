import React from 'react'
import renderer from 'react-test-renderer'

import { ErrorIndicator } from './ErrorIndicator'

describe('ErrorIndicator', () => {

  test('renders correctly', () => {
    const tree = renderer.create(
      <ErrorIndicator
        value={'OAuthException'}
        response={{
          error: {
            code: 200,
          }
        }}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders correctly when value is falsy', () => {
    const tree = renderer.create(
      <ErrorIndicator 
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

})