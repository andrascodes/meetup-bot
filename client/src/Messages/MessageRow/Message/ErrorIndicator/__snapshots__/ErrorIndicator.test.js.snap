exports[`ErrorIndicator renders correctly 1`] = `
<div
  className="ErrorIndicator">
  Error: 
  <span
    className="ErrorStyle">
    OAuthException #200
  </span>
  <i
    className="fa fa-times ErrorStyle Icon" />
</div>
`;

exports[`ErrorIndicator renders correctly when value is falsy 1`] = `<div />`;
